src/pcre2_compile.c

```diff
@@ -2766,6 +2766,60 @@ return parsed_pattern;
 
 
 
+/*************************************************
+* Maximum size of parsed_pattern for given input *
+*************************************************/
+
+/* This function is called from parse_regex() below, to determine the amount
+of memory to allocate for parsed_pattern. It is also called to check whether
+the amount of data written respects the amount of memory allocated.
+
+Arguments:
+  ptr             points to the start of the pattern
+  ptrend          points to the end of the pattern
+  utf             TRUE in UTF mode
+  options         the options bits
+
+Returns:          the number of uint32_t units for parsed_pattern
+*/
+static ptrdiff_t
```

Comment:
`size_t` ? Or PCRE2_SIZE?

Reply:
This was intentional. There is some arithmetic in the PCRE2_ASSERT lines, and some of the expressions are negative, so it made sense to make all the expressions signed rather than unsigned. Otherwise there would be ugly casts.